build_web_compilers/lib/src/dart2js_bootstrap.dart

```diff
@@ -21,23 +21,37 @@ import 'platforms.dart';
 import 'web_entrypoint_builder.dart';
 
 /// Compiles an the primary input of [buildStep] with dart2js.
+///
+/// When [generateEntrypoint] is `false` (it defaults to `true`), the generated
+/// asset is named `.dart.bootstrap.js` instead of `.dart.js`. This is used by
+/// the entrypoint builder to invoke both the dart2js and the dart2wasm
+/// compilers and generate a custom entrypoint script detecting supported
+/// browser features to load the relevant script.
 Future<void> bootstrapDart2Js(
   BuildStep buildStep,
   List<String> dart2JsArgs, {
   required bool? nativeNullAssertions,
+  bool generateEntrypoint = true,
```

Comment:
nit: maybe just have this be a more general `String outputExtension = '.dart.js'`. I would also probably make the fallback extension be `.dart.app.js` or something, and not re-use the `.dart.bootstrap.js` extension - it isn't really a bootstrap it is the actual compiled app 🤷‍♂️.

Another idea would be to allow the user to control the extension per compiler entirely. So, maybe the  `compiler` option starts allowing a map of compiler name to output extension, for the "bootstrap" js file for that compiler? If only a name of a compiler is given, then it defaults to `{ <compiler>: '.dart.js' }`.

This leaves the runtime logic of how to choose which version of the app to use more up to the user, which some users might need. And we could provide a generic JS file which can be included otherwise for trivial setups?

Reply:
I like the option of being able to customize the extension on a per-compiler basis. Making the generic JS loader script optional is also nice, since it allows for more flexibility instead of the static builder option template.

> And we could provide a generic JS file which can be included otherwise for trivial setups?

Should we enable this with the option as well? So that users that want our default loader would do:

```
  compilers:
    dart2js: '.dart2js.js'
    dart2wasm: '.dart2wasm.mjs'
    loader: '.dart.js'
```

And users who need their own entrypoint would simply remove the `loader` entry from `compilers`? This is pushing the `compilers` option a little, but if we get the default right most users hopefully don't have to worry about it too much?

Reply:
I think I would prefer to move the `loader` config out to a separate option, and not have it be a part of `compilers`? And then it can have its own sane default depending on the `compilers` configuration:

- **dart2js**: none
- **dart2wasm**: '.dart.js'
- **dart2js** _and_ **dart2wasm**: '.dart.js'

Reply:
Should we move the `dart2js_args` and `dart2wasm_args` options into the compilers map as well? (while still supporting the old mode as a fallback of course)? Then we could have:

```yaml
compilers:
  dart2js:
    extension: ".dart2js.js"
    args: ['-O4']
  dart2wasm:
    extension: ".dart.mjs"
```

I'm asking because then we should probably design values in the `compilers` map to be objects instead of strings right away, which may also be useful if we come up with other compiler-specific options later.

Reply:
That seems pretty reasonable to me